src/environment.js

```diff
@@ -18,9 +18,12 @@ function toFastProperties(obj) {
   (function () {}).prototype = obj;
 }
 
-Object.keys(Promise.prototype).filter(isNotMethod).forEach(del(Promise.prototype));
-Object.keys(Promise).filter(isNotMethod).forEach(del(Promise));
-toFastProperties(Promise);
-toFastProperties(Promise.prototype);
+const PromiseCopy = Object.assign({}, Promise);
```

Comment:
Hmm.. [This is not supported until Node 4.](https://node.green/#ES2015-built-in-extensions-Object-static-methods-Object-assign)

Is there preferred method to checking in a polyfill?